polyply/src/load_library.py

```diff
@@ -35,13 +71,17 @@ def _resolve_lib_paths(lib_names, data_path):
         list of library names
     data_path:
         location of library files
+    allowed_extensions: list[str]
+         list of allowed file extensions
     """
     files = []
     for name in lib_names:
         directory = os.path.join(data_path, name)
         for _file in os.listdir(directory):
-            _file = os.path.join(directory, _file)
-            files.append(_file)
+            _, file_extension = os.path.splitext(_file)
+            if file_extension[1:] in allowed_extensions:
+                _file = os.path.join(directory, _file)
+                files.append(_file)
```

Comment:
I'm not sure this is the right place for this. Either don't filter here, add all files to `files` irrespective of extension, and let the actual parser-selector blow up; *or* filter based on the keys in `FORCE_FIELD_PARSERS` (and either way, do some `str.casefold`)